Extension/src/LanguageServer/settings.ts

```diff
@@ -312,36 +309,134 @@ export class CppSettings extends Settings {
+    public get intelliSenseMaxMemory(): number | undefined | null { return this.getAsNumber("intelliSense.maxMemory"); }
+    public get referencesMaxConcurrentThreads(): number | undefined | null { return this.getAsNumber("references.maxConcurrentThreads"); }
+    public get referencesMaxCachedProcesses(): number | undefined | null { return this.getAsNumber("references.maxCachedProcesses"); }
+    public get referencesMaxMemory(): number | undefined | null { return this.getAsNumber("references.maxMemory"); }
+    public get codeAnalysisMaxConcurrentThreads(): number | undefined | null { return this.getAsNumber("codeAnalysis.maxConcurrentThreads"); }
+    public get codeAnalysisMaxMemory(): number | undefined | null { return this.getAsNumber("codeAnalysis.maxMemory"); }
+    public get codeAnalysisUpdateDelay(): number | undefined { return this.getAsNumber("codeAnalysis.updateDelay"); }
+    public get codeAnalysisExclude(): Excludes | undefined { return this.getAsKeyValueObject("codeAnalysis.exclude", "string", "boolean"); }
+    public get codeAnalysisRunAutomatically(): boolean | undefined { return this.getAsBoolean("codeAnalysis.runAutomatically"); }
+    public get codeAnalysisRunOnBuild(): boolean | undefined { return this.getAsBoolean("codeAnalysis.runOnBuild"); }
+    public get clangTidyEnabled(): boolean | undefined { return this.getAsBoolean("codeAnalysis.clangTidy.enabled"); }
+    public get clangTidyConfig(): string | undefined { return this.getAsString("codeAnalysis.clangTidy.config"); }
+    public get clangTidyFallbackConfig(): string | undefined { return this.getAsString("codeAnalysis.clangTidy.fallbackConfig"); }
+    public get clangTidyHeaderFilter(): string | undefined | null { return this.getAsString("codeAnalysis.clangTidy.headerFilter"); }
+    public get clangTidyArgs(): string[] | undefined { return this.getAsArrayOfStrings("codeAnalysis.clangTidy.args"); }
+    public get clangTidyUseBuildPath(): boolean | undefined { return this.getAsBoolean("codeAnalysis.clangTidy.useBuildPath"); }
+    public get clangTidyChecksEnabled(): string[] | undefined { return this.getAsArrayOfStrings("codeAnalysis.clangTidy.checks.enabled", true); }
+    public get clangTidyChecksDisabled(): string[] | undefined { return this.getAsArrayOfStrings("codeAnalysis.clangTidy.checks.disabled", true); }
+    public get clangTidyCodeActionShowDisable(): boolean | undefined { return this.getAsBoolean("codeAnalysis.clangTidy.codeAction.showDisable"); }
+    public get clangTidyCodeActionShowClear(): string { return this.getAsString("codeAnalysis.clangTidy.codeAction.showClear"); }
+    public get clangTidyCodeActionShowDocumentation(): boolean | undefined { return this.getAsBoolean("codeAnalysis.clangTidy.codeAction.showDocumentation"); }
+    public get clangTidyCodeActionFormatFixes(): boolean { return this.getAsBoolean("codeAnalysis.clangTidy.codeAction.formatFixes") ?? true; }
```

Comment:
`getAsBoolean` shouldn't return `null` or `undefined`, so the `?? true` shouldn't be necessary anymore, I think.